### PR TITLE
update root.json only when necessary

### PR DESCRIPTION
**Pull request by nexustar**
### What problem does this PR solve? 

No need to update root.json every time update something

### What is changed and how it works?

remote update of root.json before get timestamp.json, add it after get snapshot.json

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
NONE
```

### CODE REVIEW

**Comment by nexustar** [comment]
I also removed root version match check because it is not mentioned on TUF and the signature was already checked

File: pkg/repository/v1_repository.go [hunk 1]
```diff
@@ -281,6 +266,22 @@ func (r *V1Repository) updateLocalSnapshot() (*v1manifest.Snapshot, error) {
...
 			return nil, err
 		}
+
+		// Update root if needed and restart the update process
+		var oldRoot v1manifest.Root
+		_, _, err = r.local.LoadManifest(&oldRoot)
+		if err != nil {
+			return nil, err
+		}
+		newRootVersion := snapshot.Meta[v1manifest.ManifestURLRoot].Version
```
Comment (AstroProfundis) [inline pkg/repository/v1_repository.go:276]: What if the latest `snapshot.json` is signed with a new key only available in the (remote) new `root.json`?

Comment (nexustar) [reply pkg/repository/v1_repository.go:276]: fixed as new commit, it will update root.json when SignatureError or ExpirationError

**Review by AstroProfundis — APPROVED** [review]

**Pull request merged**